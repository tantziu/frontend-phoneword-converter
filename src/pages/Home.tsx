import Results from '../components/Results'
import Search from '../components/Search'
import {ChangeEvent, useState} from 'react';
import Error from '../components/Error'
import {fetchWords} from '../api/client'

const Home = () => {
    const [results, setResults] = useState([])
    const [inputNumber, setInputNumber] = useState("")
    const [validNumber, setValidNumber] = useState(false)
    // const [error, setError] = useState(null)
    // const [isLoading, setIsLoading] = useState(false)

    const onConvert = (inputNumber:string) => {
        fetchWords(inputNumber)
            .then(words => {
                setResults(words)
            })
    }   

    const onInputChange = (event:ChangeEvent) => {
        // setInputNumber((event.target as HTMLInputElement).value)
        const target= event.target as HTMLInputElement
        if (target)
            validateInput(target.value)
            setInputNumber(target.value)
    }

    const validateInput = (value:string) => {
        const regexp = /^[0-9]+$/
        regexp.test(value)
            ? setValidNumber(true)
            : setValidNumber(false)
    }

    return (
        <div className="HomePage">
            <Search input={inputNumber} validInput={validNumber} onChange={onInputChange} onConvert={onConvert}/>
            {validNumber
                ? <Results words={results}/>
                : <Error />
            }
        </div>
    )
}

export default Home

export {
    Results
}
