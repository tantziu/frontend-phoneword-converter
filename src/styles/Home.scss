@import "color";

.Home {
    text-align: center;
    
    header {
        background-color: $color-blue-1;
        padding: 10px 0px;
    }

    .content {
        width: 100%;
        margin-left: auto;
        margin-right: auto;
        padding-left: 0.25rem;
        padding-right: 0.25rem;
        
    }
}
