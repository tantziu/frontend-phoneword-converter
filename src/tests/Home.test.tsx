import React from 'react';
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import Home, {Results} from '../pages/Home'

describe('Home', () => {
    it('render without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<Home />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    test('has a valid snapshot', () => {
        const component = renderer.create(<Home/>)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})

// describe('Search', () => {
//   it('renders without crashing', () => {
//     const div = document.createElement('div')
//     ReactDOM.render(<Search/>, div)
//     ReactDOM.unmountComponentAtNode(div)
//   })
// })
describe('Results', () => {
    const props = {words: ['ad', 'ae', 'af']}

    it('render without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<Results {...props}/>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <Results {...props} />
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})