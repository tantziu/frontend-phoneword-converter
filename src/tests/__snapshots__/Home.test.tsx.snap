// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home has a valid snapshot 1`] = `
<div
  className="HomePage"
>
  <div
    className="Search"
  >
    <h3>
      Input:
    </h3>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      Convert
    </button>
  </div>
  <div
    className="table"
  >
    <h3>
      Words:
    </h3>
    <ul />
  </div>
</div>
`;

exports[`Results has a valid snapshot 1`] = `
<div
  className="table"
>
  <h3>
    Words:
  </h3>
  <ul>
    <li>
      ad
    </li>
    <li>
      ae
    </li>
    <li>
      af
    </li>
  </ul>
</div>
`;
